tuf/api/metadata.py

```diff
@@ -1085,13 +1085,15 @@ class TargetFile(BaseFile):
     Attributes:
         length: An integer indicating the length of the target file.
         hashes: A dictionary of hash algorithm names to hash values.
+        targetname: An optional string denoting the target file name.
         unrecognized_fields: Dictionary of all unrecognized fields.
     """
 
     def __init__(
         self,
         length: int,
         hashes: Dict[str, str],
+        targetname: Optional[str] = None,
```

Comment:
(this is just a drive by comment, will have to think about the justification for thw whole thing a bit still... but thanks for starting this, I think it's useful to make discussion more concrete)

What teodora said,  let's not use even more different names for the same thing... I think "targetpath" is used for this already so the attribute should probably be targetpath or path.

I believe it should not be optional though. Or is there a use case of a TargetFile without a targetpath?

Reply:
> I'm not a fan of TargetFile.targetname. How about TargetFile.filename, TargetFile.path or TargetFile.targetpath if we insist on being verbose and stick to the spec.

I had a misconception and didn't want to call it `targetpath` thinking it's not an actual path, but a file name.
Now, I read the spec and it clearly this is called `TARGETPATH`:
https://theupdateframework.github.io/specification/latest/#targetpath.

So, from both options to call it `targetpath` or `path` I would prefer `path` given it's shorter.

> I believe it should not be optional though. Or is there a use case of a TargetFile without a targetpath?

I wasn't sure when I added it as optional. I did that because it's not required by the spec, but we can use `TargetFile`
in such a way that we would always pass the path.